multiversx_sdk_cli/validators/core.py

```diff
@@ -54,7 +53,7 @@ def create_transaction_for_staking(
             rewards_address=rewards_address,
         )
 
-        if estimate_gas:
+        if not gas_limit:
```

Comment:
I thought we can drop `self.estimate_*` functions, since gas estimation is performed by controllers in sdk-py.

Reply:
Here, the transactions are not created using a controller from `sdk-py`. This will be changed in a later PR. We still left the `--gas-limit` argument in case a user wants to specify the gas.